app/src/controllers/InfuraController.js

```diff
@@ -0,0 +1,56 @@
+const ObservableStore = require('obs-store')
+const extend = require('xtend')
+const log = require('loglevel')
+
+// const POLLING_INTERVAL = 10 * 60 * 1000
+const POLLING_INTERVAL = 3000
+
+export default class InfuraController {
+  constructor(opts = {}) {
+    const initState = extend(
+      {
+        infuraNetworkStatus: {}
+      },
+      opts.initState
+    )
+    this.store = new ObservableStore(initState)
+    log.info('Polling network status...')
+  }
+
+  async checkNetworkStatus() {
+    window.addEventListener('online', () =>
+      this.store.updateState({
+        infuraNetworkStatus: true
+      })
```

Comment:
clean up the listener here. Use the following helper method
https://github.com/torusresearch/torus-embed/blob/feat/popup-unblock/src/embedUtils.js#L36
https://github.com/torusresearch/torus-embed/blob/feat/popup-unblock/src/embed.js#L750